certauth.go

```diff
@@ -77,6 +80,22 @@ func (a *Auth) Handler(h http.Handler) http.Handler {
 	})
 }
 
+// RouterHandler implements the httprouter.Handle for integration with github.com/julienschmidt/httprouter
+func (a *Auth) RouterHandler(h httprouter.Handle) httprouter.Handle {
```

Comment:
Ok, I think this would work if your handlers do not use the httprouter's additional Params arg. Also, you could no longer use httprouter's `httprouter.GET()` pattern (which is the canonical example they use).

eg: this works:

```
func HelloWithoutParams(w http.ResponseWriter, r *http.Request) {
        fmt.Fprintf(w, "hello without params")
}

..
        router := httprouter.New()
        router.GET("/", Index)
        router.Handler("GET", "/hello/:name", auth.Handler(http.HandlerFunc(HelloWithoutParams)))
```

I am not a big fan of this.  1. It looks like shit. 3 levels of wrapping is cognitive overload. 2. You lose httprouter's Params.

If we're going to recommend httprouter is a recommended path, then why neuter this library to only work with lowest-common denominator of net/http?